src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -307,18 +321,33 @@ bool CSVFormatReader::readField(
         /// they do not contain empty unquoted fields, so this check
         /// works for tuples as well.
         column.insertDefault();
-        return false;
     }
-
-    if (format_settings.null_as_default && !isNullableOrLowCardinalityNullable(type))
+    else if (current_row_has_missing_fields)
+    {
+        column.insertDefault();
+    }
+    else if (format_settings.null_as_default && !isNullableOrLowCardinalityNullable(type))
     {
         /// If value is null but type is not nullable then use default value instead.
-        return SerializationNullable::deserializeTextCSVImpl(column, *buf, format_settings, serialization);
+        res = SerializationNullable::deserializeTextCSVImpl(column, *buf, format_settings, serialization);
+    }
+    else
+    {
+        /// Read the column normally.
+        serialization->deserializeTextCSV(column, *buf, format_settings);
+        res = true;
     }
 
-    /// Read the column normally.
-    serialization->deserializeTextCSV(column, *buf, format_settings);
-    return true;
+    if (is_last_file_column && format_settings.csv.ignore_extra_columns)
+    {
+        // Skip all fields to next line.
+        while (checkChar(format_settings.csv.delimiter, *buf))
```

Comment:
Here may be the left chars in buf is not start with `format_settings.csv.delimiter`, then the extra columns may can not be skipped.

Reply:
They should have entered the value of the previous field. Can you give an example?

Reply:
Whitespace before delimiter:
```
avogar-dev :) select * from format(CSV, 'a UInt32', ' 42 , 42') settings input_format_csv_missing_as_default=1, input_format_csv_ignore_extra_columns=1

Received exception:
Code: 117. DB::Exception: Expected end of line: (at row 1)
: Could not print diagnostic info because two last rows aren't in buffer (rare case)
: While executing ParallelParsingBlockInputFormat. (INCORRECT_DATA)
```

Reply:
Fixed (added skipWhitespacesAndTabs)

Reply:
> They should have entered the value of the previous field. Can you give an example?

In my tests, I have meet such a situation: if the table is defined like `test_tbl(a String, b String, c Date)`, And the input row maybe `abcde, abcded, abde, abcdefed`,  `c` column value can not be deseraized to `Date` type, but no exception will be throwed, it will directly take 10 chars to deseralized include the delimiter char,  and the left chars in the buffer would be `efed`. 

The exception is like blow
```
Caused by: java.lang.RuntimeException: Cannot read DateTime: unexpected word: 
Row 1:
Column 0,   name: a, type: Nullable(Int32),         parsed text: "1"
Column 1,   name: b, type: Nullable(String),        parsed text: "abs"
Column 2,   name: c, type: Nullable(Date32),        parsed text: "2022-03-02"
Column 3,   name: d, type: Nullable(DateTime64(6)), parsed text: "12434455,fal"ERROR
Code: 41. DB::ParsingException: Cannot read DateTime: unexpected word. (CANNOT_PARSE_DATETIME) (version 23.6.1.1)

```
@kardymonds

Reply:
@KevinyhZou I think this is for performance reasons, readDateTextImpl takes 10 characters from the buffer (YYYY-MM-DD) and the external buffer in such a not quite consistent state. To fix this, it looks like you need to change the parsing of the date from the buffer.